### DISCUSSION TITLE
fdtransfer error: FdTransfer accept(): Resource temporarily unavailable

### DISCUSSION THREAD

**Discussion by evanzhang87** [async-profiler/async-profiler#680] [Q&A] [answered]
I try to profile Java in a container, but I got two problems.
1.
```
[root@xxxxx async-profiler-2.8.3-linux-x64]# ./profiler.sh --fdtransfer 94953
Failed to inject profiler into 94953
/bin/true: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /xxx/sync-profile/async-profiler-2.8.3-linux-x64/build/libasyncProfiler.so)
/bin/true: /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /xxx/sync-profile/async-profiler-2.8.3-linux-x64/build/libasyncProfiler.so)
/bin/true: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /xxx/sync-profile/async-profiler-2.8.3-linux-x64/build/libasyncProfiler.so)
/bin/true: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /xxx/sync-profile/async-profiler-2.8.3-linux-x64/build/libasyncProfiler.so)
```
I use latest code to build `libasyncProfiler.so`, and build with image `egymgmbh/jdk11-builder:latest`, It works well in my ubuntu20 5.15.0-52-generic, but it not works in centos7 5.10.56.

2.
```
FdTransfer accept(): Resource temporarily unavailable
```
I noticed that `make sure that the target container can access libasyncProfiler.so by the same absolute path as on the host.`, but I can't restart container to bind a new volume,so I use docker cp, but it not works

**Comment by apangin** [comment, marked as answer]
You build async-profiler on one system, but try to use it on another with an older version of libstdc++. This will not work.

There are prebuilt binaries on the [Releases](https://github.com/jvm-profiling-tools/async-profiler/releases) page - they are built using the most compatible toolchain and should work pretty much everywhere. I do not support third party ad hoc builds, especially since the problem is not specific to async-profiler. It's a common issue that has been discussed all over internet many times.

**Reply by evanzhang87** [reply]
thanks
